app/logic/events.py

```diff
@@ -75,9 +76,14 @@ def deleteEventAndAllFollowing(eventId):
         if event:
             if event.recurringId:
                 recurringId = event.recurringId
-                recurringSeries = list(Event.select().where((Event.recurringId == recurringId) & (Event.startDate >= event.startDate)))
-        for seriesEvent in recurringSeries:
-            seriesEvent.delete_instance(recursive = True)
+                recurringSeries = list(Event.select(Event.id).where((Event.recurringId == recurringId) & (Event.startDate >= event.startDate)))
+                #session['lastDeletedEvent'] = []
+                deletedEventList = []                
+                for recurringEvent in recurringSeries:
+                    deletedEventList.append(recurringEvent.id)
+                Event.update({Event.deletionDate: datetime.now()}).where((Event.recurringId == recurringId) & (Event.startDate >= event.startDate)).execute()
+                Event.update({Event.deletedBy: g.current_user}).where((Event.recurringId == recurringId) & (Event.startDate >= event.startDate)).execute()
+                return deletedEventList
 
 def deleteAllRecurringEvents(eventId):
```

Comment:
Let's have this function just call the previous one, with the first event in the series.